warden/x/async/keeper/keeper.go

```diff
@@ -146,3 +168,109 @@ func (k Keeper) GetFutureVotes(ctx context.Context, futureId uint64) ([]types.Fu
 
 	return votes, nil
 }
+
+func (k Keeper) futureReadyCallback(
+	ctx context.Context,
+	id uint64,
+	output []byte,
+) error {
+	future, err := k.futures.Get(ctx, id)
+
+	if err != nil {
+		return err
+	}
+
+	if future.Callback == "" {
+		return nil
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	abi, err := callbacks.IAsyncCallbackMetaData.GetAbi()
+	method := "cb"
+	if _, ok := abi.Methods[method]; !ok {
+		return fmt.Errorf("invalid callback method: %v", method)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	cbAddress, err := precommon.AddressFromBech32Str(future.Callback)
+
+	if err != nil {
+		return err
+	}
+
+	data, err := abi.Pack(method, id, output)
+	if err != nil {
+		return err
+	}
+
+	res, err := k.CallEVMWithData(
+		sdkCtx,
+		common.BytesToAddress(k.asyncModuleAddress.Bytes()),
+		&cbAddress,
+		data,
+	)
+
+	if res.Failed() {
+		// Do not throw error if contract fails
+		return nil
+	}
+
+	return err
+}
+
+func (k Keeper) CallEVMWithData(
```

Comment:
let's not export this 

```suggestion
func (k Keeper) callEVMWithData(
```